infra-cdk/lib/stacks/IAMStack.ts

```diff
@@ -17,18 +20,34 @@ export class IAMStack extends cdk.Stack {
     constructor(scope: Construct, id: string, props: IAMStackProps) {
         super(scope, id, props);
 
+         /** Setted up the option to choose between S3 vs DynamoDB or both */
+         const statements = [];
+
+         if (props.tableName) {
+            // Allow access to DynamoDB table
+            statements.push(new iam.PolicyStatement({
+                actions: ['dynamodb:Scan', 'dynamodb:GetItem', 'dynamodb:PutItem'],
+                resources: [
+                    `arn:aws:dynamodb:${ENVIRONMENT.region}:${ENVIRONMENT.account}:table/${props.tableName}`,
+                    `arn:aws:dynamodb:${ENVIRONMENT.region}:${ENVIRONMENT.account}:table/${props.reportsTableName}`,
+                ],
+            }));
+        }
+
+        if (props.bucketStack) {
+            // Allow access to S3 bucket
+            statements.push(new iam.PolicyStatement({
```

Comment:
Nice job setting up these permissions :) It's tricky but it looks like you got it all right!